File: SDSI.py
import time
def get_ds_iter(n):
    while len(n) > 1:
        n = str(sum(int(d) for d in n))
    return n

inp = input("Enter number: ")
if inp.isdigit():
    start = time.perf_counter()
    res = get_ds_iter(inp)
    print(f"Result: {res} | Time: {(time.perf_counter()-start)*1000:.6f} ms")

File: SDSI.java
import java.util.Scanner;

public class SDSI {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter number: ");
        String s = sc.next();
        if(s.matches("\\d+")) {
            long start = System.nanoTime();
            while (s.length() > 1) {
                long sum = 0;
                for (char c : s.toCharArray()) sum += (c - '0');
                s = String.valueOf(sum);
            }
            long end = System.nanoTime();
            System.out.println("Result: " + s + " | Time: " + (end-start)/1000000.0 + " ms");
        }
    }
}
